extension/src/background/index.ts

```diff
@@ -171,11 +173,32 @@ window.signLoan = async () => {
     // on the pop-up matches what is stored in the extension's
     // storage. It would be better to send around the swap ID to check
     // that the wallet is signing the same transaction the user has authorised
-    signLoan(walletName)
-        .then(resolveLoanSignRequest)
-        .catch(rejectLoanSignRequest)
-        .then(cleanupPendingLoan);
+
+    // if we receive an error, we respond directly, else we return the details
+    return await signLoan(walletName).catch(rejectLoanSignRequest);
```

Comment:
`return await` is unnecessary, you can just remove the `await` and return the promise directly.